### PR TITLE
activesupport: update `to_xml` signature to use an optional block

### PR DESCRIPTION
**Pull request by hjwylde**
Split from #735 to be smaller. Type signature updates for invalid signatures in activesupport, due to blocks being marked as required, but they should be optional.

### CODE REVIEW

File: gems/activesupport/6.0/activesupport.rbs [hunk 1]
```diff
@@ -394,6 +394,83 @@ end
...
+  #     <message>
+  #       <created-at>2008-03-07T09:58:18+01:00</created-at>
+  #       <id>1</id>
+  #       <name>1</name>
+  #       <updated-at>2008-03-07T09:58:18+01:00</updated-at>
+  #       <user-id>1</user-id>
+  #     </message>
+  #   </messages>
+  #
+  def to_xml: (?::Hash[untyped, untyped] options) ?{ (untyped) -> untyped } -> untyped
```
Comment (tk0miya) [inline gems/activesupport/6.0/activesupport.rbs:473]: Note:
https://github.com/rails/rails/blob/v6.0.6.1/activesupport/lib/active_support/core_ext/array/conversions.rb#L209

File: gems/activesupport/6.0/activesupport.rbs [hunk 2]
```diff
@@ -427,6 +504,70 @@ end
...
+  #       "DateTime"   => "dateTime",
+  #       "Time"       => "dateTime"
+  #     }
+  #
+  # By default the root node is "hash", but that's configurable via the <tt>:root</tt> option.
+  #
+  # The default XML builder is a fresh instance of <tt>Builder::XmlMarkup</tt>. You can
+  # configure your own builder with the <tt>:builder</tt> option. The method also accepts
+  # options like <tt>:dasherize</tt> and friends, they are forwarded to the builder.
+  def to_xml: (?::Hash[untyped, untyped] options) ?{ (untyped) -> untyped } -> untyped
```
Comment (tk0miya) [inline gems/activesupport/6.0/activesupport.rbs:570]: Note:
https://github.com/rails/rails/blob/v6.0.6.1/activesupport/lib/active_support/core_ext/hash/conversions.rb#L90

**Review by tk0miya — COMMENTED** [review]
APPROVE

File: gems/activesupport/6.0/activesupport.rbs (hunk 1 above)
Comment (hjwylde) [reply gems/activesupport/6.0/activesupport.rbs:473]: Thanks! Just to clarify, are these notes just for references to the relevant code, or review points?

Comment (hjwylde) [reply gems/activesupport/6.0/activesupport.rbs:473]: After reading the other reviews I believe they are just references, so I'll merge this one too, thank you.

**Pull request merged**